app/assets/javascripts/oxalis/view/action-bar/share_modal_view.js

```diff
@@ -47,17 +45,26 @@ function Hint({ children, style }) {
 
 class ShareModalView extends PureComponent<ShareModalProp, State> {
   state = {
-    isPublic: false,
+    isPublic: this.props.isPublic,
+    sharingToken: "",
   };
 
-  componentWillReceiveProps(newProps: ShareModalProp) {
-    this.setState({ isPublic: newProps.isPublic });
+  componentDidMount() {
+    this.fetch();
+  }
+
+  async fetch() {
+    const { name, owningOrganization } = this.props.dataset;
+    const datasetId = { name, owningOrganization };
+    const sharingToken = await getDatasetSharingToken(datasetId);
+    this.setState({ sharingToken });
   }
 
   getUrl() {
-    const loc = window.location;
-    const { pathname } = loc;
-    const url = `${loc.origin + pathname}${loc.hash}`;
+    const { location } = window;
+    const { pathname, origin, hash } = location;
+    const query = this.props.dataset.isPublic ? "" : `?token=${this.state.sharingToken}`;
```

Comment:
Hmm, on a second thought, I'm wondering whether it makes sense to always add the dataset sharing token. Contrived scenario:

- user A creates a tracing for a dataset D
- user B is in the same organization as A, but doesn't have access to D
- user A shares a link to his tracing, but uses the "private" setting
- user B can see the dataset, even though user A might have assumed that **only users with "explicit" access to D can view the tracing**

Maybe too contrived. On a certain level, this can probably be considered equal to the entire dataset token mechanism. Users with access to a private dataset can always weaken the "privacy" of a dataset.

If we "wanted" to be more pedantic, we could do:

- tracing is only visible to users which belong to the same organization and have access to the dataset (don't append token)
- tracing is only visible to users which belong to the same organization (append token)
- tracing is visible to everybody (append token)

Reply:
I think that's just something that we'll have to decide. For the sake of simplicity I would opt to stick with the two options instead of three, and add a warning/info that this will share the dataset with others, even if they usually don't have access. Maybe @fm3 has an opinion about this as well?

Reply:
I’d also tend to go for the simpler version with a warning/info. Not a strong opinion, though.